spmd/tensor/api.py

```diff
@@ -236,7 +285,7 @@ def redistribute(
         return Redistribute.apply(self, device_mesh, placements)
 
     def local_tensor(self) -> torch.Tensor:
-        return self._local_tensor  # type: ignore
+        return ToTorchTensor.apply(self)  # pyre-ignore[16]: autograd func
```

Comment:
Thanks for making this happening. Can we move the dt.grad.local_tensor to dt.local_tensor().grad? So that FSDP + TP can work. Or we need to create an ad-hoc backward hook?

Reply:
If you do `dt = DistributedTensor.from_local(foo)` then the gradients will be populated on `foo.grad`.
Neither `dt.grad` nor `dt.local_tensor()` will have a .grad field.

Reply:
I see. Sounds like this is what we want for TP + FSDP. Let me try it out to verify that. Thanks!

Reply:
Yeah, that means we probably need to invent a API in TP + FSDP integration, called `distributed_fsdp_param`, where it use `from_local` instead of the plain `Tensor` constructor. 

The `distribute_tensor` API likely won't use from_local for this case as module parameters should be dist param